### PR TITLE
feat: setup entire login and course purchase setup

### LINKED ISSUE #169: Add purchases, payment information
Right now, a lot of the backend (user auth and course purchases) is deligated to appx
We want to create a strategy where courses can be bought in house
This will require some schema design so if you're thiniking of working on this please post the schema and approach first

### PR DESCRIPTION
**Pull request by devsargam**
Resolves #169 

This pr currently adds:
- [x] Ability to signup in house
- [x] Hash Password
- [x] In house sign in
- [x] Ability to purchase a course in house
- [x] Change the schema for storing purchases
- [x] Integrate Razorpay 

In future:
- [x] ~Verify the user with some method~ Not required for v0

### CODE REVIEW

**Comment by devsargam** [comment]
hey @hkirat, how should I proceed with mail verification? I can implement it my way but what would suit you the most???

**Comment by hkirat** [comment]
No need to verify emails for v0

**Comment by devsargam** [comment]
hey @hkirat, Do we need to support login with appx as a fallback method?

**Comment by devsargam** [comment]
Also, just to be sure, content is still comming from appx right?

**Comment by hkirat** [comment]
No all content is in house currently as well
appx gives us
1. authentication (they store user information)
2. Let us know which user has purchased which course - ref - https://github.com/code100x/cms/blob/main/src/utiles/appx.ts#L15

**Comment by devsargam** [comment]
oh, makes a lot sense. Thanks

**Comment by devsargam** [comment]
https://github.com/code100x/cms/assets/76874341/a27839ce-12b4-415c-a390-6154c35f09d1

**Comment by devsargam** [comment]
@hkirat ready for review. Could you check this out?

File: src/utiles/getContentForUser.ts [hunk 1]
```diff
@@ -54,42 +51,5 @@ export async function getPurchases(email: string) {
     };
   });
 
-  if (LOCAL_CMS_PROVIDER) {
-    return courses;
-  }
-
-  const baseUrl = `${APPX_BASE_API}/get/checkemailforpurchase`;
```
Comment (devsargam) [inline src/utiles/getContentForUser.ts:61]: Removed the entire appx dependency from the project

**Comment by devsargam** [comment]
As this is v0 a lot of things to be improved. LMK your thoughts on this and I shall continue.

PR already getting a lot large and messy.

**Comment by devsargam** [comment]
@hkirat hoping it comes to your notifs

**Comment by hkirat** [comment]
Will review it asap

File: src/utiles/getContentForUser.ts (hunk 1 above)
Comment (hkirat) [reply src/utiles/getContentForUser.ts:61]: hmm this might be tricky we need some backward compatability for the current courses

**Comment by devsargam** [comment]
Just clarifying a few things
I have left a bunch of TODOs which can be their own issue. 
Simple enough to fix not totally necessary but can make the app a lot better and performant
not totally sure thooo

File: src/utiles/getContentForUser.ts (hunk 1 above)
Comment (devsargam) [reply src/utiles/getContentForUser.ts:61]: yep that was what I was basically worried about but seems doable. 
I am here to put every ounce of effort for this pr just let me know!!!!

**Comment by devsargam** [comment]
Also, bit of clarification required. Can we quickly hop on a meet just to clarify things a bit ofc if u are free thooo

**Comment by devsargam** [comment]
hey @hkirat, what are u thoughts about this pr? Is this even somewhat doable?

**Comment by hkirat** [comment]
too many merge confs, gonna close for now

**Comment by hkirat** [comment]
checking

**Pull request closed without merging**